nucypher/crypto/kits.py

```diff
@@ -106,6 +108,26 @@ def sender(self, enrico):
     def __bytes__(self):
         return super().to_bytes(include_alice_pubkey=True)
 
+    def ensure_correct_sender(self,
+                              enrico: Optional["Enrico"] = None,
+                              policy_encrypting_key: Optional[UmbralPublicKey] = None):
+        """
+        Make sure that the sender of the message kit is set and corresponds to
+        the given ``enrico``, or create it from the given ``policy_encrypting_key``.
+        """
+        if self.sender:
+            if enrico and self.sender != enrico:
+                raise ValueError
```

Comment:
Perhaps a error message here:
```suggestion
                raise ValueError("Unexpected sender {enrico}")
```

or something of the sort...?

Reply:
Added a message here and in another branch as well.